paasta_tools/chronos_tools.py

```diff
@@ -471,14 +471,21 @@ def create_complete_config(service, job_name, soa_dir=DEFAULT_SOA_DIR):
     )
 
     complete_config['name'] = compose_job_id(service, job_name)
+
+    # resolve conflicts between the 'desired_state' and soa_configs disabled
+    # flag.
     desired_state = chronos_job_config.get_desired_state()
```

Comment:
Why not put this logic in `get_desired_state` so that every caller of it will get it?